File: server/Dockerfiles/Dockerfile.java
# ./server/Dockerfiles/Dockerfile.java
# Use OpenJDK 17 as the base image for Java
FROM openjdk:11

# Set the working directory inside the container to /usr/src/app
WORKDIR /usr/src/app

# Copy the Java source code from the host machine to the container's /usr/src/app directory
COPY ./server/java-files/Main.java /usr/src/app/


# Compile the Java program using javac
RUN javac Main.java

# Set the default command to run the Java program
CMD ["java", "Main"]

File: server/Dockerfiles/Dockerfile.cpp
# ./server/Dockerfiles/Dockerfile.cpp
# Start from a base image with GCC installed
FROM gcc:latest

# Set the working directory inside the container
WORKDIR /usr/src/app

# Copy the C++ source code from the host machine to the container
COPY ./server/cpp-files/main.cpp /usr/src/app/


# Compile the C++ program using g++
RUN g++ -o main main.cpp

# Set the default command to run the compiled program
CMD ["./main"]
